lib/generators/solidus_stripe/install/templates/config/initializers/solidus_stripe.rb

```diff
@@ -13,6 +13,29 @@
   # config.webhook_signature_tolerance = 150
 end
 
+# Allow sending the payment id in the payment form and avoid the automatic
+# creation of an additional payment by Spree::PaymentCreate.
+#
+# We're providing a inline module to make it easier to remove the mokey-patch
+# once the upstream PR is merged and generally available.
+#
+# Fixes https://github.com/solidusio/solidus/issues/2680
+# Fixed by https://github.com/solidusio/solidus/pull/4909
+begin
+  module SolidusStripe::PaymentCreateWithExistingPayment
+    def build
+      @payment ||= order.payments.find(attributes[:id]) if attributes[:id]
```

Comment:
With this whole picture in mind, I had some thoughts about how we can upstream this to core. What about marking some payment methods as "accepting multiple attempts", maybe with some overridable methods similar to the existing `support_profiles?`. When it's `true` (and Stripe will be) core will act differently taking the already existing payment of the same type, instead of passing the id explicitly, which opens the door to extra security implications.

Reply:
> I tried to think this through and the only issue I can find is that [amount is a permitted parameter](https://github.com/solidusio/solidus/blob/master/core/lib/spree/permitted_attributes.rb#L72), potentially allowing amount updates

That's serious enough 🙂

Reply:
This should be now addressed, the PaymentCreate patch will return early if the `:id` parameter is provided, so no update of any sort will happen.